source/h2_frames.c

```diff
@@ -20,23 +20,65 @@
 
 #include <aws/common/logging.h>
 
+#include <aws/io/stream.h>
+
 #include <inttypes.h>
 
-/* #TODO: Don't raise AWS_H2_ERR_* enums, raise AWS_ERROR_* .
- *        Also, if encoder raises error corresponding to AWS_H2_ERR, should
- *        we send that code in the GOAWAY, or always treat encoder errors as AWS_H2_ERR_INTERNAL?
- *        Like, you're only supposed to inform peer of errors that were their fault, right? */
+#define ENCODER_LOGF(level, encoder, text, ...)                                                                        \
+    AWS_LOGF_##level(AWS_LS_HTTP_ENCODER, "id=%p " text, (encoder)->logging_id, __VA_ARGS__)
+
+#define ENCODER_LOG(level, encoder, text) ENCODER_LOGF(level, encoder, "%s", text)
 
 const struct aws_byte_cursor aws_h2_connection_preface_client_string =
     AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n");
 
+/* Initial values and bounds are from RFC-7540 6.5.2 */
+const uint32_t aws_h2_settings_initial[AWS_H2_SETTINGS_END_RANGE] = {
+    [AWS_H2_SETTINGS_HEADER_TABLE_SIZE] = 4096,
+    [AWS_H2_SETTINGS_ENABLE_PUSH] = 1,
+    [AWS_H2_SETTINGS_MAX_CONCURRENT_STREAMS] = UINT32_MAX, /* "Initially there is no limit to this value" */
+    [AWS_H2_SETTINGS_INITIAL_WINDOW_SIZE] = 65535,
```

Comment:
That's what I was doing? you'd access them like so:
`aws_h2_settings_initial[AWS_H2_SETTINGS_HEADER_TABLE_SIZE]`

instead of:
`AWS_H2_HEADER_TABLE_SIZE_INITIAL`

I thought it would be more "programmable" to have them in an array.
I'll change it in the future if it's just a pain to read